src/test/java/au/com/dmg/fusion/MessageTest.java

```diff
@@ -139,13 +159,25 @@ public void fromJsonTestBadEnums() {
ssageClass\":\"Service\",\"MessageCategory\":\"Payment\",\"MessageType\":\"Request\",\"ServiceID\":\"09f58793-8f91-4aab-953f-bed4c8770a02\",\"SaleID\":\"VA POS\",\"POIID\":\"DMGVA001\"},\"PaymentRequest\":{\"CustomFields\":[{\"key\":\"testCustomFieldKey\",\"type\":\"object\",\"value\":\"{\\\"TransitData\\\":{\\\"DriverID\\\":123,\\\"OperatorID\\\":456,\\\"ContractID\\\":\\\"0f9653cc-a68b-11ed-afa1-0242ac120002\\\",\\\"VehicleID\\\":789,\\\"RouteVariant\\\":\\\"X\\\",\\\"TransactionLocation\\\":{\\\"Lattitude\\\":33.8688,\\\"Longitude\\\":151.2093},\\\"Trip\\\":{\\\"Boarding\\\":{\\\"StopID\\\":\\\"\\\",\\\"StopName\\\":\\\"\\\",\\\"ZoneID\\\":\\\"\\\"},\\\"Destination\\\":{\\\"StopID\\\":\\\"\\\",\\\"StopName\\\":\\\"\\\",\\\"ZoneID\\\":\\\"\\\"}},\\\"Ticket\\\":[{\\\"Type\\\":\\\"Adult\\\",\\\"Price\\\":1.1,\\\"ID\\\":\\\"9fe0e990-a68d-11ed-afa1-0242ac120002\\\"}]}}\"},{\"key\":\"testCustomFieldKey2\",\"type\":\"object\",\"value\":\"{\\\"TransitData2\\\":{\\\"DriverID\\\":123,\\\"OperatorID\\\":456,\\\"ContractID\\\":\\\"0f9653cc-a68b-11ed-afa1-0242ac120002\\\",\\\"VehicleID\\\":789,\\\"RouteVariant\\\":\\\"X\\\",\\\"TransactionLocation\\\":{\\\"Lattitude\\\":33.8688,\\\"Longitude\\\":151.2093},\\\"Trip\\\":{\\\"Boarding\\\":{\\\"StopID\\\":\\\"\\\",\\\"StopName\\\":\\\"\\\",\\\"ZoneID\\\":\\\"\\\"},\\\"Destination\\\":{\\\"StopID\\\":\\\"\\\",\\\"StopName\\\":\\\"\\\",\\\"ZoneID\\\":\\\"\\\"}},\\\"Ticket\\\":[{\\\"Type\\\":\\\"Adult\\\",\\\"Price\\\":1.1,\\\"ID\\\":\\\"9fe0e990-a68d-11ed-afa1-0242ac120002\\\"}]}}\"}],\"SaleData\":{\"OperatorID\":\"OperatorID\",\"OperatorLanguage\":\"en-us\",\"ShiftNumber\":\"1\",\"SaleTransactionID\":{\"TransactionID\":\"8ea678a7-eeb0-4977-9f8d-011897a6eda7\",\"TimeStamp\":\"2023-02-11T01:24:04.165Z\"},\"SaleReferenceID\":\"pos\",\"SaleTerminalData\":{\"TerminalEnvironment\":\"Attended\",\"SaleCapabilities\":[\"CashierStatus\",\"CashierError\",\"CashierInput\",\"CustomerAssistance\",\"PrinterReceipt\"]},\"TokenRequestedType\":\"Transaction\"},\"PaymentTransaction\":{\"AmountsReq\":{\"Currency\":\"AUD\",\"RequestedAmount\":5.95,\"CashBackAmount\":0},\"SaleItem\":[{\"CustomFields\":[{\"key\":\"SampleProductCustomFieldKey\",\"type\":\"integer\",\"value\":\"21\"}],\"ProductCode\":\"9300675079655\",\"UnitOfMeasure\":\"ea\",\"Quantity\":\"1\",\"UnitPrice\":\"3.95\",\"ItemAmount\":3.95,\"ProductLabel\":\"Coke\"}]},\"PaymentData\":{\"PaymentType\":\"Normal\",\"PaymentInstrumentData\":{\"PaymentInstrumentType\":\"Card\"}}},\"SecurityTrailer\":{\"ContentType\":\"id-ct-authData\",\"AuthenticatedData\":{\"Version\":\"v0\",\"Recipient\":{\"KEK\":{\"Version\":\"v4\",\"KEKIdentifier\":{\"KeyIdentifier\":\"SpecV1TestMACKey\",\"KeyVersion\":\"20191122164326.594\"},\"KeyEncryptionAlgorithm\":{\"Algorithm\":\"des-ede3-cbc\"},\"EncryptedKey\":\"9ECF3D39C698C43944E3C41D25D63BCE\"},\"MACAlgorithm\":{\"Algorithm\":\"id-retail-cbc-mac-sha-256\"},\"EncapsulatedContent\":{\"ContentType\":\"id-data\"},\"MAC\":\"07B337E17467B44A\"}}}}}");
```

Comment:
Please make the sample data more generic like the original test message for _fromJsonTestBadEnums_.  

Also, like the original _fromJsonTestBadEnums_ content wherein it has a test case for _getUnitOfMeasure() == UnitOfMeasure.Other_ and then _getUnitOfMeasure() == UnitOfMeasure.Centimetre_, I would recommend that you create a test case for _getCustomFields().get(0).getType() == CustomFieldType.Unknown_ in addition to the getCustomFields().get(0).getType() == CustomFieldType.object test case that you added.